bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1128 @@
+                return (Clip2BridgeHandler) handler;
+            }
+        }
+        throw new AssetNotLoadedException("Bridge handler missing");
+    }
+
+    /**
+     * Do a double lookup to get the cached resource that matches the given ResourceType.
+     *
+     * @param resourceType the type to search for.
+     * @return the Resource, or null if not found.
+     */
+    private @Nullable Resource getCachedResource(ResourceType resourceType) {
+        String commandResourceId = commandResourceIds.get(resourceType);
+        return Objects.nonNull(commandResourceId) ? serviceContributorsCache.get(commandResourceId) : null;
+    }
+
+    /**
+     * Return a ResourceReference to this handler's resource.
+     *
+     * @return a ResourceReference instance.
+     */
+    public ResourceReference getResourceReference() {
+        return new ResourceReference().setId(resourceId).setType(thisResource.getType());
+    }
+
+    /**
+     * Register the 'DynamicsAction' service.
+     */
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(DynamicsActions.class);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command commandParam) {
+        if (RefreshType.REFRESH.equals(commandParam)) {
+            if ((thing.getStatus() == ThingStatus.ONLINE) && updateDependenciesDone) {
+                ScheduledFuture<?> task = updateServiceContributorsTask;
+                if (Objects.isNull(task) || !task.isDone()) {
+                    updateServiceContributorsTask = scheduler.schedule(() -> {
+                        try {
+                            updateServiceContributors();
+                        } catch (ApiException | AssetNotLoadedException e) {
+                            // exceptions will not occur here since thing is already online
+                        }
+                    }, 3, TimeUnit.SECONDS);
+                }
+            }
+            return;
+        }
+
+        Channel channel = thing.getChannel(channelUID);
+        if (channel == null) {
+            logger.warn("{} -> handleCommand() channelUID:{} does not exist", resourceId, channelUID);
+            return;
+        }
+
+        ResourceType lightResourceType = thisResource.getType() == ResourceType.DEVICE ? ResourceType.LIGHT
+                : ResourceType.GROUPED_LIGHT;
+
+        Resource putResource = null;
+        String putResourceId = null;
+        Command command = commandParam;
+        String channelId = channelUID.getId();
+        Resource cache = getCachedResource(lightResourceType);
+
+        switch (channelId) {
+            case CHANNEL_2_ALERT:
+                putResource = Setters.setAlert(new Resource(lightResourceType), command, cache);
+                scheduler.schedule(() -> updateState(channelUID, new StringType(ActionType.NO_ACTION.name())), 10,
+                        TimeUnit.SECONDS);
```

Comment:
This task needs to be cancelled in `dispose()`, otherwise you risk this when thing is disposed shorter than 10 seconds after this: _Handler Clip2ThingHandler tried updating the thing status although the handler was already disposed._

Reply:
Ok. Done.